sdi/transport.go

```diff
@@ -0,0 +1,76 @@
+package sdi
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+)
+
+// Transport defines interface for http client transport methods
+type Transport interface {
+	SetTransport(t *http.Transport)
+	WithContext(ctx context.Context)
+	RoundTrip(r *http.Request) (*http.Response, error)
+}
+
+// DefaultTransport default configuration for HTTP communications
+type DefaultTransport struct {
+	HTTPTransport *http.Transport
+	Context       context.Context
+}
+
+// SetTransport sets the underlying HTTPTransport value
+func (s *DefaultTransport) SetTransport(t *http.Transport) {
+	s.HTTPTransport = t
```

Comment:
I'd expect `HTTPTransport` to be a private attribute if it has a method to be configured.